contracts/src/Zappers/Modules/Exchanges/HybridCurveUniV3Exchange.sol

```diff
@@ -0,0 +1,203 @@
+
+        // Uniswap
+        ISwapRouter uniV3RouterCached = uniV3Router;
+
+        USDC.approve(address(uniV3RouterCached), curveUsdcAmount);
+
+        // See: https://docs.uniswap.org/contracts/v3/guides/swaps/multihop-swaps
+        bytes memory path;
+        if (address(WETH) == address(collToken)) {
+            path = abi.encodePacked(USDC, feeWethColl, WETH);
+        } else {
+            path = abi.encodePacked(USDC, feeUsdcWeth, WETH, feeWethColl, collToken);
+        }
+
+        ISwapRouter.ExactInputParams memory params = ISwapRouter.ExactInputParams({
+            path: path,
+            recipient: msg.sender,
+            deadline: block.timestamp,
+            amountIn: curveUsdcAmount,
+            amountOutMinimum: _minCollAmount
+        });
+
+        // Executes the swap.
+        uint256 uniV3CollAmount = uniV3RouterCached.exactInput(params);
+
+        // return leftovers to user
+        _returnLeftovers(initialBalances);
+
+        return uniV3CollAmount;
+    }
+
+    // Optionally Coll -> WETH, and WETH -> USDC on UniV3; then USDC -> Bold on Curve
+    function swapToBold(uint256 _collAmount, uint256 _minBoldAmount) external returns (uint256) {
+        InitialBalances memory initialBalances;
+        _setHybridExchangeInitialBalances(initialBalances);
+
+        // Uniswap
+        ISwapRouter uniV3RouterCached = uniV3Router;
+
+        collToken.approve(address(uniV3RouterCached), _collAmount);
+
+        // See: https://docs.uniswap.org/contracts/v3/guides/swaps/multihop-swaps
+        bytes memory path;
+        if (address(WETH) == address(collToken)) {
+            path = abi.encodePacked(WETH, feeUsdcWeth, USDC);
+        } else {
+            path = abi.encodePacked(collToken, feeWethColl, WETH, feeUsdcWeth, USDC);
+        }
+
+        ISwapRouter.ExactInputParams memory params = ISwapRouter.ExactInputParams({
+            path: path,
+            recipient: msg.sender,
+            deadline: block.timestamp,
+            amountIn: _collAmount,
+            amountOutMinimum: 0
+        });
+
+        // Executes the swap.
+        uint256 uniV3UsdcAmount = uniV3RouterCached.exactInput(params);
+
+        // Curve
+        ICurveStableswapNGPool curvePoolCached = curvePool;
+        USDC.approve(address(curvePoolCached), uniV3UsdcAmount);
+
+        uint256 boldAmount =
+            curvePoolCached.exchange(int128(USDC_INDEX), int128(BOLD_TOKEN_INDEX), uniV3UsdcAmount, _minBoldAmount);
+        boldToken.transfer(msg.sender, boldAmount);
+
+        // return leftovers to user
+        _returnLeftovers(initialBalances);
+
+        return boldAmount;
+    }
+
+    function _setHybridExchangeInitialBalances(InitialBalances memory initialBalances) internal view {
+        initialBalances.tokens[0] = boldToken;
+        initialBalances.tokens[1] = USDC;
+        initialBalances.tokens[2] = WETH;
+        initialBalances.tokens[3] = collToken;
```

Comment:
I guess it doesn't hurt, but seems redundant to add WETH twice, if it's the collToken of the branch.